torchvision/models/maxvit.py

```diff
@@ -0,0 +1,595 @@
+
+        # stem
+        self.stem = nn.Sequential(
+            Conv2dNormActivation(
+                input_channels, stem_channels, 3, stride=2, norm_layer=None, activation_layer=None, bias=True
+            ),
+            Conv2dNormActivation(
+                stem_channels, stem_channels, 3, stride=1, norm_layer=None, activation_layer=None, bias=True
+            ),
+        )
+
+        # account for stem stride
+        input_size = (input_size[0] // 2, input_size[1] // 2)
+
+        # blocks
+        self.blocks = nn.ModuleList()
+        in_channels = [stem_channels] + block_channels[:-1]
+        out_channels = block_channels
+
+        # precompute the stochastich depth probabilities from 0 to stochastic_depth_prob
+        # since we have N blocks with L layers, we will have N * L probabilities uniformly distributed
+        # over the range [0, stochastic_depth_prob]
+        p_stochastic = np.linspace(0, stochastic_depth_prob, num=sum(block_layers)).tolist()
+
+        p_idx = 0
+        for in_channel, out_channel, num_layers in zip(in_channels, out_channels, block_layers):
+            self.blocks.append(
+                MaxVitBlock(
+                    in_channels=in_channel,
+                    out_channels=out_channel,
+                    squeeze_ratio=squeeze_ratio,
+                    expansion_ratio=expansion_ratio,
+                    normalization_fn=normalization_fn,
+                    activation_fn=activation_fn,
+                    head_dim=head_dim,
+                    mlp_ratio=mlp_ratio,
+                    mlp_dropout=mlp_dropout,
+                    attn_dropout=attn_dropout,
+                    partition_size=partition_size,
+                    input_grid_size=input_size,
+                    n_layers=num_layers,
+                    p_stochastic=p_stochastic[p_idx : p_idx + num_layers],
+                ),
+            )
+            input_size = self.blocks[-1].grid_size
+            p_idx += num_layers
+
+        self.classifier = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1),
+            nn.Flatten(),
+            nn.Linear(block_channels[-1], out_classes, bias=False),
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.stem(x)
+        for block in self.blocks:
+            x = block(x)
+        x = self.classifier(x)
+        return x
+
+
+def max_vit_T_224(num_classes: int) -> MaxVit:
+    return MaxVit(
+        input_channels=3,
+        stem_channels=64,
+        input_size=(224, 224),
+        out_classes=num_classes,
+        block_channels=[64, 128, 256, 512],
+        block_layers=[2, 2, 5, 2],
+        stochastic_depth_prob=0.2,
+        squeeze_ratio=0.25,
+        expansion_ratio=4.0,
+        normalization_fn=nn.BatchNorm2d,
+        activation_fn=GeluWrapper,
+        head_dim=32,
+        mlp_ratio=2,
+        mlp_dropout=0.0,
+        attn_dropout=0.0,
+        partition_size=7,
+    )
+
+
+def max_vit_S_224(num_classes: int) -> MaxVit:
```

Comment:
I recommend dropping all other variants (and expected files) from here until we have fully trained the tiny version. It will reduce the amount of boilerplate code we have to review.